lib/libc/Kconfig

```diff
@@ -7,18 +7,39 @@
 #
 
 menu "C Library"
-depends on !NATIVE_APPLICATION
+
+choice LIBC_IMPLEMENTATION
+	prompt "C Library Implementation"
+	default MINIMAL_LIBC
+
+config MINIMAL_LIBC
+	bool "Minimal c library"
+	depends on !NATIVE_APPLICATION
+	depends on !SOC_SERIES_STM32H7X
+	help
+	  Build with minimal C library.
 
 config NEWLIB_LIBC
-	bool "Build with newlib c library"
+	bool "Newlib c library"
+	depends on !NATIVE_APPLICATION
 	help
 	  Build with newlib library. The newlib library is expected to be
 	  part of the SDK in this case.
 
+config NATIVE_LIBC
```

Comment:
Instead of this can we have a
`config EXTERNAL_LIBC` (or USER_PROVIDED_LIBC or whatever other name) to indicate neither of the 2 libc's provided by Zephyr is used?
That way we cover all 3rd party libCs. (There doesn't seem to be a need, at least yet, to differentiate the host libC from any other 3rd party libC)

Note the `STDOUT_CONSOLE` depend would need to remain on `!NATIVE_APPLICATION`
It would also make a lot of sense to have the libc cmake file not be included on that option instead of NATIVE_APP.